web3/_utils/contract_error_handling.py

```diff
@@ -134,11 +113,43 @@ def raise_contract_logic_error_on_revert(response: RPCResponse) -> RPCResponse:
         # on the ContractCustomError exception
         raise ContractCustomError(data, data=data)
 
-    # Geth case:
-    if message_present and response["error"].get("code", "") == 3:
-        raise ContractLogicError(message, data=data)
-    # Geth Revert without error message case:
-    if message_present and "execution reverted" in message:
-        raise ContractLogicError("execution reverted", data=data)
+
+def raise_contract_logic_error_on_revert(response: RPCResponse) -> RPCResponse:
+    """
+    Reverts contain a `data` attribute with the following layout:
+        "Reverted "
```

Comment:
Given the above, some note here that the "Reverted" may or may not be present, while the rest of the layout is standard.